packages/dev/core/src/ShadersWGSL/iblShadowsIcdfx.fragment.fx

```diff
@@ -0,0 +1,43 @@
+#define PI 3.1415927
+varying vUV: vec2f;
+
+var cdfx: texture_2d<f32>;
+
+fn fetchCDF(x: u32) -> f32 {
+    return textureLoad(cdfx,  vec2u(x, 0), 0).x;
+}
+
+fn bisect(size: u32, targetValue: f32) -> f32
+{
+    var a: u32 = 0;
+    var b = size - 1;
+    while (b - a > 1) {
+        var c: u32 = (a + b) >> 1;
+        if (fetchCDF(c) < targetValue) {
+            a = c;
+        }
+        else {
+            b = c;
+        }
+    }
+    return mix( f32(a),  f32(b), (targetValue - fetchCDF(a)) / (fetchCDF(b) - fetchCDF(a))) /  f32(size - 1);
+}
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+    var cdfSize: vec2u = textureDimensions(cdfx, 0);
+    var cdfWidth: u32 = cdfSize.x;
+    var icdfWidth: u32 = cdfWidth - 1;
+    var currentPixel: vec2u =  vec2u(input.position.xy);
```

Comment:
shader preprocessor can't handle this expression see [here](https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts#L280)
```suggestion
    var currentPixel: vec2u =  vec2u(fragmentInputs.position.xy);
```